src/Controllers/UserController.js

```diff
@@ -60,11 +60,18 @@ export class UserController extends AdaptableController {
       updateFields._email_verify_token_expires_at = {__op: 'Delete'};
     }
 
-    return this.config.database.update('_User', query, updateFields).then((document) => {
-      if (!document) {
-        throw undefined;
-      }
-      return Promise.resolve(document);
+    var self = this;
+    var checkIfAlreadyVerified = new RestQuery(self.config, Auth.master(self.config), '_User', {username: username, emailVerified: true});
```

Comment:
"I don't think it's excessive responsibility on the verifyEmail" - so this is incorrect.  

It should be in resend email because you're resending an email to the same email address, so it makes sense to not resend it if it's already verified, if a user that has already verified a previous email, changed the email, the expectation of the system is that it can change the email and call this method.  Now your changes is preventing that correct expectation from executing.  This breakage of existing expectation is a product of introducing responsibilities to functions that shouldn't have those responsibilities.